File: test/org.eclipse.vjet.test.core.jstojava/src/org/eclipse/vjet/dsf/jst/validation/vjo/samples/dox/ebay/vjoPro/vjoPro4javadev/samples/vjlib/ImageAnimateImage.java
/*******************************************************************************
 * Copyright (c) 2005-2012 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/

package org.eclipse.vjet.dsf.jst.validation.vjo.samples.dox.ebay.vjoPro.vjoPro4javadev.samples.vjlib;




import java.util.List;

import org.eclipse.vjet.dsf.jsgen.shared.ids.FieldProbIds;
import org.eclipse.vjet.dsf.jsgen.shared.validation.vjo.VjoSemanticProblem;
import org.eclipse.vjet.dsf.jst.validation.vjo.VjoValidationBaseTester;
import org.junit.Before;
import org.junit.Test;




/**
 * ImageAnimateImage.java
 * 
 * @author <a href="mailto:liama@ebay.com">liama</a>
 * @since JDK 1.5
 */
//@Category( { P3, FAST, UNIT })
//@ModuleInfo(value="DsfPrebuild",subModuleId="JsToJava")
public class ImageAnimateImage extends VjoValidationBaseTester {

    @Before
    public void setUp() {
        expectProblems.clear();
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 36, 0));
        expectProblems.add(createNewProblem2(FieldProbIds.UndefinedField, 20, 0));
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 25, 0));
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 33, 0));
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 22, 0));
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 36, 0));
        expectProblems
                .add(createNewProblem2(FieldProbIds.UndefinedField, 24, 0));
    }

    @Test
    //@Category( { P3, FAST, UNIT })
    //@Description("Test VJO Sample project, To validate false positive ")
    // The bug 6110 still exist. now we update the test js
    // file via alias. if use alias it's work.
    public void testImageAnimateImage() {
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                VjoValidationBaseTester.VJLIB_FOLDER,
                "dox.ebay.vjoPro.vjoPro4javadev.samples.vjlib.",
                "ImageAnimateImage.js", this.getClass());
        assertProblemEquals(expectProblems, problems);
    }
}

File: core/test/org.eclipse.vjet.test.core.jstojava/vjLibTestFiles/dox/ebay/vjoPro/vjoPro4javadev/samples/vjlib/ImageAnimateImage.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
vjo.ctype("dox.ebay.vjoPro.vjoPro4javadev.samples.vjlib.ImageAnimateImage")
.needs("vjoPro.dsf.utils.Object")
.needs('vjoPro.dsf.document.Image','A')
.needs('vjoPro.samples.images.Image','B')
.protos({

//> public void constructs(String psId, String [] paImageUrls)
constructs : function (psId, paImageUrls) {
if (document.images){// object detection
this.sId = psId;
// add image
this.aImageUrls = paImageUrls;
// preload image
for(var i = 0; i < this.aImageUrls.length; ++i) {
this.vj$.B.preload(this.aImageUrls[i]);
}
}
},

//> public void animate()
animate:function(){
if (document.images){// object detection
if (this.iCurrentIndex > this.aImageUrls.length - 1) {
this.iCurrentIndex = 0;
}
this.vj$.A.load(this.sId, this.aImageUrls[this.iCurrentIndex++]);
setTimeout(vjoPro.dsf.utils.Object.hitch(this,"animate"),  500);
}
},
iCurrentIndex : 0 //<Number
})
.endType();
